modules/geolocation/geolocation.py

```diff
@@ -129,3 +140,210 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+
+    def convert_input(self, latitude: int, longitude: int, altitude: int, worldOrigin: np.ndarray, gpsOffset: np.ndarray, cameraOffset: np.ndarray, eulerCamera: dict, eulerPlane: dict):
+        """
+        Converts telemtry data into workable data
+
+        Parameters
+        ----------
+        latitude: int
+            GPS latitude of the plane
+        longitude: int
+            GPS longitude of the plane
+        altitude: int
+            GPS altitude of the plane
+        worldOrigin: ndarray
+            Coordinates of the origin of the world space
+        gpsOffset: ndarray
+            Offset from GPS module to origin of plane space
+        cameraOffset: ndarray
+            Offset from origin of plane space to camera
+        eulerCamera : dict
+            dictionary containing euler angles for the camera:
+                - roll, pitch and yaw rotations with the keys 'x', 'y' and 'z' respectively
+        eulerPlane : dict
+            dictionary containing euler angles for the plane:
+                - roll, pitch and yaw rotations with the keys 'x', 'y' and 'z' respectively
+        Returns
+        -------
+        list:
+            list of numpy arrays that represent the o, c, u and v vectors
+        """
+
+        # get plane and camera rotation matrices
+        planeRotation = self.__calculate_rotation_matrix(eulerPlane)
+        cameraRotation = self.__calculate_rotation_matrix(eulerCamera)
+
+        # get plane and camera compound rotation matrix
+        # note: assume apply plane -> camera rotations in that order
```

Comment:
Isn't it the other way around? The code is also different from this comment.

Reply:
i thought it should be plane rotation first then camera rotation since the camera euler angles are wrt plane space. so you'd need to transform into plane space first before applying the camera rotation?

either way, yes, the comment doesn't match the code

Reply:
Update this comment.